test/gateway/rewrite-headers.js

```diff
@@ -0,0 +1,101 @@
+'use strict'
```

Comment:
maybe I'm missing it, but I don't see where this file tests that rewriteHeaders is actually called and, more importantly, that its return value is actually used.

Reply:
Tests to `rewriteHeaders` are at line 64 and 65